frontend/ui/widget/focusmanager.lua

```diff
@@ -159,7 +162,63 @@ function FocusManager:_verticalStep(dy)
 end
 
 function FocusManager:getFocusItem()
+    if not self.layout then
+        return nil
+    end
     return self.layout[self.selected.y][self.selected.x]
 end
 
+function FocusManager:sendTapEventToFocusedWidget()
+    local focused_widget = self:getFocusItem()
+    if focused_widget then
+        -- center of widget position
+        local point = focused_widget.dimen:copy()
+        point.x = point.x + point.w / 2
+        point.y = point.y + point.h / 2
+        point.w = 0
+        point.h = 0
+        UIManager:sendEvent(Event:new("Gesture", {
+            ges = "tap",
+            pos = point,
+        }))
+    end
+    return focused_widget
+end
```

Comment:
Clever. :+1:

Reply:
Clever indeed :)
Beware most widgets dont get their self.dimen.x and y updated (at paintTo() time). FrameContainer is one of the rare few that do.
(Never dared updating all of them, with my philosophy of "change less, break less"... so, sometimes, when I need these x/y, I just wrap my thing in a FrameContainer with no border/padding... :/)